app/javascript/app/components/autocomplete-search/autocomplete-search.js

```diff
@@ -0,0 +1,46 @@
+import { createElement } from 'react';
+import Proptypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import { deburrUpper } from 'app/utils';
+
+import AutocompleteSearchComponent from './autocomplete-search-component';
+import actions from './autocomplete-search-actions';
+import { getFilteredCountriesWithPath } from './autocomplete-search-selectors';
+
+export { default as component } from './autocomplete-search-component';
+export { initialState } from './autocomplete-search-reducers';
+export { default as reducers } from './autocomplete-search-reducers';
+export { default as styles } from './autocomplete-search-styles';
+export { default as actions } from './autocomplete-search-actions';
+
+const mapStateToProps = (state) => {
+  const { query } = state.autocompleteSearch;
+  return {
+    query: deburrUpper(query),
+    searchList: getFilteredCountriesWithPath(state.autocompleteSearch)
+  };
+};
+
+const AutocompleteSearchContainer = (props) => {
+  const onOptionClick = (selected) => {
```

Comment:
To avoid bugs I'm going to move this to a component so we can use the react life cycle and in the component will unmount the query will be removed.